lib/publiccloud/basetest.pm

```diff
@@ -30,7 +31,10 @@ sub provider_factory {
 
     $args{provider} //= get_required_var('PUBLIC_CLOUD_PROVIDER');
 
-    if ($args{provider} eq 'EC2') {
+    if (get_var('PUBLIC_CLOUD_EXISTING_ADDRESS')) {
```

Comment:
I would rename it to `PUBLIC_CLOUD_INSTANCE_IP` because it is the address of an instance, not the address of public cloud.

```suggestion
    if (get_var('PUBLIC_CLOUD_INSTANCE_IP')) {
```